mount.c

```diff
@@ -253,13 +265,22 @@ void mount_entries(const char *root, const struct mount_entry *mounts, size_t nm
 			};
 
 			for (size_t i = 0; i < lengthof(symlinks); ++i) {
-				const char *path = makepath("%s%s/%s", root, mnt->target, symlinks[i].path);
+				const char *path = makepath("%s%s/%s", root, mnttarget, symlinks[i].path);
 
 				if (symlink(symlinks[i].target, path) == -1 && errno != EEXIST) {
 					err(1, "mount_entries: bst_devtmpfs: symlink %s -> %s",
 							symlinks[i].target, path);
 				}
 			}
+
+			const char *real_target = makepath("%s%s", root, mnt->target);
```

Comment:
I might use `makepath_r` in this instance since you're passing it into `do_mount()`.

Reply:
I'll just pass /dev directly in do_mount